robot/basestation/app.py

```diff
@@ -98,8 +103,99 @@ def stream():
 def initialSection():
     return '0'
 
-@app.route("/ping_rover")
-def ping_rover():
+@app.route('/navigation/inputTemplates/goal-buttons/<count>', methods= ["GET"])
+def goal_Buttons(count):
+   return flask.render_template("elements/navigation/htmlTemplates/goal/goal-buttons.html", count =  count)
+
+@app.route('/navigation/inputTemplates/new-goal-coordinates-btn/<goalId>', methods=["GET"])
+def new_Goal_Button(goalId):
+   return flask.render_template("elements/navigation/htmlTemplates/goal/new-goal-coordinates-btn.html", goalId = goalId)
+
+@app.route('/navigation/inputTemplates/antenna-DD/<target>' , methods = ["GET"])
+def antenna_DD(target):
+   return flask.render_template("elements/navigation/htmlTemplates/antenna/antenna-DD-input-template.html", target = target)
+         
+@app.route('/navigation/inputTemplates/antenna-DDM/<target>' , methods = ["GET"])
+def antenna_DDM(target):
+   return flask.render_template("elements/navigation/htmlTemplates/antenna/antenna-DDM-input-template.html", target = target)
+
+@app.route('/navigation/inputTemplates/antenna-DMS/<target>' , methods = ["GET"])
+def antenna_DMS(target):
+   return flask.render_template("elements/navigation/htmlTemplates/antenna/antenna-DMS-input-template.html", target = target)
+
+@app.route('/navigation/inputTemplates/goal-DD/<target>/<count>' , methods = ["GET"])
+def goal_DD(target,count):
+   return flask.render_template("elements/navigation/htmlTemplates/goal/goal-DD-input-template.html", target = target, count = count)
+
+
+@app.route('/navigation/inputTemplates/goal-DDM/<target>/<count>' , methods = ["GET"])
+def goal_DDM(target,count):
+   return flask.render_template("elements/navigation/htmlTemplates/goal/goal-DDM-input-template.html", target = target, count = count)
+
+@app.route('/navigation/inputTemplates/goal-DMS/<target>/<count>' , methods = ["GET"])
+def goal_DMS(target,count):
+    return flask.render_template("elements/navigation/htmlTemplates/goal/goal-DMS-input-template.html", target = target,  count = count)
```

Comment:
So the whole front end is pretty much fine, but the persistence is not so great. We were thinking that instead of storing objects on the server, we can just publish them on the ros network. There would need to be a node that knows about these goals, waypoints etc and that just publishes it every second or so, updating all the operators at once